nearcore/src/config.rs

```diff
@@ -458,11 +466,16 @@ impl Default for Config {
             view_client_throttle_period: default_view_client_throttle_period(),
             trie_viewer_state_size_limit: default_trie_viewer_state_size_limit(),
             max_gas_burnt_view: None,
+            num_epochs_to_keep_store_data: default_num_epochs_to_keep_store_data(),
         }
     }
 }
 
 impl Config {
+    pub fn num_epochs_to_keep_store_data(&self) -> u64 {
+        max(MIN_NUM_EPOCHS_TO_KEEP_STORE_DATA, self.num_epochs_to_keep_store_data)
```

Comment:
I would validate and fail starting the node instead of silently ignoring the value from the config. WDYT?

Reply:
@frol Yes, I agree. Where should the validation be done, maybe something like this:

    pub fn num_epochs_to_keep_store_data(&self) -> u64 {
        if self.num_epochs_to_keep_store_data < MIN_NUM_EPOCHS_TO_KEEP_STORE_DATA {
              panic!("num_epochs_to_keep_store_data: {} is below minimum: {}, 
                        self.num_epochs_to_keep_store_data,
                        MIN_NUM_EPOCHS_TO_KEEP_STORE_DATA)
        }
        self.num_epochs_to_keep_store_data
   }

Reply:
@pmnoxx this is a weird place to do the validation. It should be done on deserialization/read-from-file/constructor stage

Reply:
@frol How about inside:
```

    pub fn from_file(path: &Path) -> Self {
        let mut file = File::open(path)
            .unwrap_or_else(|_| panic!("Could not open config file: `{}`", path.display()));
        let mut content = String::new();
        file.read_to_string(&mut content).expect("Could not read from config file.");
        Config::from(content.as_str())
    }
```

Reply:
@frol I updated the PR to do it inside `from_file`